images/rag/embedding.py

```diff
@@ -38,73 +41,70 @@ def get_eval_results(key, eval_results):
+    parser.add_argument("-p", "--port", help="Vector DB port")
+    parser.add_argument("-a", "--auth", help="Authentication headers per vector DB requirements")
     parser.add_argument("-n", "--collection-name", help="Collection name in vector DB")
     parser.add_argument("-f", "--folder", help="Plain text folder path")
-    parser.add_argument("-m", "--model",   default="local:BAAI/bge-base-en", help="LLM model used for embeddings [local,llama2, or any other supported by llama_index]")
-    parser.add_argument("-e", "--include-evaluation",   default="True", help="preform evaluation [True/False]")
-    parser.add_argument("-q", "--question-folder",   default="", help="docs folder for questions gen")
-    parser.add_argument("-c", "--chunk",   default="1500", help="chunk size for embedding")
-    parser.add_argument("-l", "--overlap",   default="10", help="chunk overlap for embedding")
-    parser.add_argument("-o", "--output", help="persist folder")
+    parser.add_argument("-m", "--model",   default="local:BAAI/bge-base-en", help="LLM model used for embeddings [local, llama2, or any other supported by llama_index]")
+    parser.add_argument("-e", "--include-evaluation",   default="True", help="Perform evaluation [True/False]")
+    parser.add_argument("-q", "--question-folder",   default="", help="Docs folder for questions gen")
+    parser.add_argument("-c", "--chunk",   default="1500", help="Chunk size for embedding")
+    parser.add_argument("-l", "--overlap",   default="10", help="Chunk overlap for embedding")
+    parser.add_argument("-o", "--output", help="Vector DB output folder")
+
 
+    # execute
+    args = parser.parse_args()
 
-    # execute 
-    args = parser.parse_args() 
-    
     PERSIST_FOLDER = args.output
     CHUNK_SIZE=int(args.chunk)
     CHUNK_OVERLAP=int(args.overlap)
-    
-    # setup storage context 
-    if args.vector_type == "local": 
-        print("** Local embeddings ") 
-        storage_context = StorageContext.from_defaults()
-        
-    elif args.vector_type == "chromadb":
-        print("** chromadb embeddings ")
-        #Validate Inputs 
-        if args.url is None or args.port is None: 
-            print(" Missing URL or PORT ")
-            return 
-        
-        chroma_client = chroma.HttpClient(host=args.url, port=args.port, headers=json.loads(args.auth) )
-        collection = chroma_client.create_collection(
-            name=args.collection_name,
-            get_or_create = True            
-            )
-        vector_store = ChromaVectorStore(chroma_collection=collection)
-        storage_context = StorageContext.from_defaults(vector_store=vector_store)
-        
-        
-    elif args.vector_type == "chromadb-local":
-        chroma_client = chromadb.Client()
-        collection = chroma_client.create_collection(
-            name=args.collection_name,
-            get_or_create = True            
+
+    # setup storage context
+    match args.vector_type:
```

Comment:
Note that `case` doesn't support Python below 3.11. 
 will need to ensure our base image is using the correct python version

Reply:
We're good: https://github.com/RHEcosystemAppEng/lightspeed-rag-pipeline/blob/0b6a707d818b248d755075ce994d7da9fa8a2273/images/rag/dockerfile#L1